zbus/src/address/transport/unixexec.rs

```diff
@@ -0,0 +1,107 @@
+use std::ffi::OsString;
+use std::fmt::Display;
+use std::os::unix::ffi::OsStrExt;
+use std::os::unix::process::CommandExt;
+use std::path::PathBuf;
+use std::process::Stdio;
+
+use tokio::net::UnixStream;
+use tracing::warn;
+
+use crate::Error;
+
+use super::encode_percents;
+
+/// A unixexec domain socket transport in a D-Bus address.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct UnixExec {
+    pub(super) path: PathBuf,
+    pub(super) arg0: Option<OsString>,
+    pub(super) args: Vec<String>,
+}
+
+impl UnixExec {
+    /// Create a new unixexec transport with the given path and arguments.
+    pub fn new(path: PathBuf, arg0: Option<OsString>, args: Vec<String>) -> Self {
+        Self { path, arg0, args }
+    }
+
+    pub(super) fn from_options(opts: std::collections::HashMap<&str, &str>) -> crate::Result<Self> {
+        let Some(path) = opts.get("path") else {
+            return Err(crate::Error::Address(
+                "unixexec address is missing `path`".to_owned(),
+            ));
+        };
+
+        let arg0 = opts.get("argv0").map(OsString::from);
+
+        let mut args: Vec<String> = Vec::new();
+        let mut arg_index = 1;
+        while let Some(arg) = opts.get(format!("argv{arg_index}").as_str()) {
+            args.push(arg.to_string());
+            arg_index += 1;
+        }
+
+        Ok(Self::new(PathBuf::from(path), arg0, args))
+    }
+
+    pub(super) async fn connect(self) -> crate::Result<UnixStream> {
```

Comment:
We can support connection? According to the specs, this is not a connectable address type. :thinking:

Reply:
If I'm reading the spec right, it's the other way around (connectable, but not listenable):
> unixexec addresses are connectable, but are not listenable. ([source](https://dbus.freedesktop.org/doc/dbus-specification.html#transports-exec))

In any case, the current implementation does seem to work as expected and can be tested with minimal changes (https://github.com/vially/zbus/commit/9f8037cde9401b3ed3e67017d18d6fad88c0c9ab) to the `watch-systemd-jobs` example. It even seems to work when connecting to a *remote* D-Bus session over SSH.

Reply:
> If I'm reading the spec right, it's the other way around (connectable, but not listenable):

Oh, you're absolute right. Sorry, I was very tired when I read it. :)

> > unixexec addresses are connectable, but are not listenable. ([source](https://dbus.freedesktop.org/doc/dbus-specification.html#transports-exec))
> 
> In any case, the current implementation does seem to work as expected and can be tested with minimal changes ([vially@9f8037c](https://github.com/vially/zbus/commit/9f8037cde9401b3ed3e67017d18d6fad88c0c9ab)) to the `watch-systemd-jobs` example. It even seems to work when connecting to a _remote_ D-Bus session over SSH.

Awesome! Thanks so much again. Once you've addressed other comments, we can quickly merge it hopefully.